main.js

```diff
@@ -50,18 +50,20 @@ savedPalettesSection.addEventListener('click', async function(event) {
     }
 });
 
-function getPromiseFromEvent(element, listenerName) {
+function getPromiseFromEvent(event) {
+
     return new Promise(function (resolve) {
+        event.addEventListener('click', listener);
+
         function listener(event) {
             if (event.target.classList.contains('modal-exit-button')) {
                 shouldDelete = false;
-            } else if (event.target.nodeName === 'BUTTON') {
+                resolve(event);
+            } else if (event.target.classList.contains('delete-palette-button')) {
                 shouldDelete = true;
+                resolve(event);
             }
-            element.removeEventListener(listenerName, listener);
-            resolve(event);
         };
```

Comment:
i think we can still have `resolve(event)` just a single time where it was before?

Reply:
If `resolve(event)` is outside the conditional then you can click anywhere on the box to close it. Having it in each of the conditions resolves it for that condition and you have to either click the button or the X. So resolving the event basically only closes the modal and what is inside the condition actually makes the delete happen or not by assigning the boolean to the variable `shouldDelete`

Reply:
wow, thanks for the great explanation makes total sense!